[ahmadsyuaib] iP

ahmadsyuaib opened the pull request:
# Stobberi
>“An unorganized plan is just a dream; a checklist makes it a reality.” – Somebody Probably ([source](https://chat.openai.com/))

Stobberi frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And most importantly, it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing **YOU**
- [ ] Reminders (coming soon)

If you are a Java programmer or simply learning how to use Java, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        new Stobberi().run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ahmadsyuaib committed:
Added greetings

ahmadsyuaib committed:
added Echo

ahmadsyuaib committed:
added AddList

ahmadsyuaib committed:
formatted Code

ahmadsyuaib committed:
added checklists

ahmadsyuaib committed:
created different task types

ahmadsyuaib committed:
Created TextUITesting

ahmadsyuaib committed:
Added error exceptions

ahmadsyuaib committed:
Added delete

ahmadsyuaib committed:
cleaned the code a bit

ahmadsyuaib committed:
update constants naming

ahmadsyuaib committed:
remove static methods

ahmadsyuaib committed:
add new file list.txt

ahmadsyuaib committed:
Add new file list.txt

ahmadsyuaib committed:
Create new file ListSaver.java

ahmadsyuaib committed:
add getList to get saved list of tasks

ahmadsyuaib committed:
add save

ahmadsyuaib committed:
Merge branch 'branch-Level-7'

ahmadsyuaib committed:
remove unnecessary print statement

ahmadsyuaib committed:
add new imports

ahmadsyuaib committed:
add DateTime feature to tasks

ahmadsyuaib committed:
add error for start time after end time for events

ahmadsyuaib committed:
add ability to check deadlines and events on certain dates

ahmadsyuaib committed:
Merge branch 'branch-Level-8'

ahmadsyuaib committed:
rename listSaver to Storage

ahmadsyuaib committed:
rename ListSaver to Storage

ahmadsyuaib committed:
Merge branch 'branch-Storage'

ahmadsyuaib committed:
create Ui file

ahmadsyuaib committed:
extracted greetings

ahmadsyuaib committed:
extract display format into Ui

ahmadsyuaib committed:
Merge branch 'branch-Ui'

ahmadsyuaib committed:
extract TaskList

ahmadsyuaib committed:
Merge branch 'branch-TaskList'

ahmadsyuaib committed:
create Command files for each executable action

ahmadsyuaib committed:
clean unnecessary comments

ahmadsyuaib committed:
create package for exceptions

ahmadsyuaib committed:
add all classes into one Stobberi package

ahmadsyuaib committed:
arrange files into packages

ahmadsyuaib committed:
arrange all classs into respective folders

ahmadsyuaib committed:
add Junit items

ahmadsyuaib committed:
create Javadoc for TaskList class

ahmadsyuaib committed:
create Javadoc for Ui

ahmadsyuaib committed:
create Javadoc for Storage class

ahmadsyuaib committed:
improve Javadoc for Ui class

ahmadsyuaib committed:
Create javadoc for Ui class

ahmadsyuaib committed:
clean switch case

ahmadsyuaib committed:
create new method that checks if the description has the input word

ahmadsyuaib committed:
create filter by word

ahmadsyuaib committed:
change hasTrue to hasWord

ahmadsyuaib committed:
edit the list interface

ahmadsyuaib committed:
create command for find

ahmadsyuaib committed:
create option for find to be a command

ahmadsyuaib committed:
clean code

ahmadsyuaib committed:
add javadoc

ahmadsyuaib committed:
add Javadoc

ahmadsyuaib committed:
add more Javadoc

ahmadsyuaib committed:
add Javadoc

ahmadsyuaib committed:
create javadoc

ahmadsyuaib committed:
Create Javadoc

ahmadsyuaib committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  create option for find to be a command
  create command for find
  edit the list interface
  change hasTrue to hasWord
  create filter by word
  create new method that checks if the description has the input word

ahmadsyuaib committed:
Merge branch 'branch-A-JavaDoc'

ahmadsyuaib committed:
Merge branch 'branch-A-CodingStandard'

ahmadsyuaib committed:
Clean code

ahmadsyuaib committed:
Create Javadoc

ahmadsyuaib committed:
Add more Javadoc

ahmadsyuaib committed:
rename package to lowercase

ahmadsyuaib committed:
add line before displaying String

ahmadsyuaib committed:
Create a new test case

ahmadsyuaib committed:
Merge branch 'branch-JUnit'

* branch-JUnit:
  Create a new test case

ahmadsyuaib committed:
add A-Jar

weijianwong commented on src/main/java/stobberi/command/Command.java:
```diff
@@ -0,0 +1,45 @@
+ * Represents a command that can be executed. The command has an exit flag that indicates
+ * whether the command should terminate the application or not.
+ */
+public class Command {
```
I like how you created Command as a parent class and created subclasses for different commands that all have the execute function. This makes the code much cleaner and there will not be a long switch statement as you are using polymorphism to deal with the different commands. Nice job! Maybe you could declare this class as an Abstract class?

weijianwong commented on src/main/java/stobberi/command/Command.java:
```diff
@@ -0,0 +1,45 @@
+ * Represents a command that can be executed. The command has an exit flag that indicates
+ * whether the command should terminate the application or not.
+ */
+public class Command {
```
I like how you created Command as a parent class and created subclasses for different commands that all have the execute function. This makes the code much cleaner and there will not be a long switch statement as you are using polymorphism to deal with the different commands. Nice job! Maybe you could declare this class as an Abstract class?

ahmadsyuaib replied:
Noted. Thank you. I agree that it should be an Abstract class as there should not be an instance of this class.

weijianwong commented on src/main/java/stobberi/Stobberi.java:
```diff
@@ -0,0 +1,68 @@
+import stobberi.components.TaskList;
+import stobberi.components.Ui;
+
+/**
```
Great job leaving class and method header javadoc comments. They are clear and concise and they effectively convey the information of the class or method to the reader.

ahmadsyuaib replied:
Thank you!

weijianwong commented on src/main/java/stobberi/components/Ui.java:
```diff
@@ -0,0 +1,58 @@
+    /**
+     * Displays a farewell message to the user.
+     */
+    public void goodbye() {
```
Perhaps the name for this function could changed to a more action-oriented verb such as "sayGoodbye" or "displayGoodbyeMessage"?

ahmadsyuaib replied:
I agree. I have made it more action-oriented so that others understand the method easier.

weijianwong commented on src/main/java/stobberi/components/Parser.java:
```diff
@@ -0,0 +1,82 @@
+        String command = parts[0];
+        String restOfCommand = String.join(" ", java.util.Arrays.copyOfRange(parts, 1, parts.length));
+
+        switch (command) {
```
Good job with the indentation of the switch statement. This is a common error for many so great job for avoiding it!

ahmadsyuaib replied:
Thank you!

weijianwong commented on src/main/java/stobberi/components/Parser.java:
```diff
@@ -0,0 +1,82 @@
+package stobberi.components;
```
Package name is in lower cases, class names are in PascalCase and variable names are in camelCase. Great job following the naming convention throughout!

ahmadsyuaib replied:
Thank you!

weijianwong reviewed (commented):
Great job overall with with the naming conventions and following all the coding style guidelines. I really liked how you did the command class and created different classes for different commands with the execute function. That was very clean!

KennyLewi commented on src/main/java/stobberi/components/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+    /**
+     * Displays the last added task.
+     */
+    public void displayLastAdded() {
```
maybe you can rename this to displayLastAddedTask to make it a clearer method name?

ahmadsyuaib replied:
That is true. I have edited it accordingly.

KennyLewi commented on src/main/java/stobberi/task/Task.java:
```diff
@@ -0,0 +1,89 @@
+     *
+     * @return {@code true} if the task is done, {@code false} otherwise.
+     */
+    public boolean getDone() {
```
maybe rename this as isDone?

ahmadsyuaib replied:
That is true since it is a boolean. Thank you so much.

KennyLewi commented on src/main/java/stobberi/Stobberi.java:
```diff
@@ -0,0 +1,68 @@
+ * @version 1.0
+ * @since 2024-08-29
+ */
+public class Stobberi {
```
good job on naming classes as nouns!

ahmadsyuaib replied:
Thank you!

KennyLewi commented on src/main/java/stobberi/components/Storage.java:
```diff
@@ -0,0 +1,119 @@
+ * It supports reading tasks of different types (Todo, Deadline, Event) from a file and saving tasks
+ * back to the file. This class is responsible for the persistence of task data.
+ */
+public class Storage {
```
Good job on using clear English names for all your class and method names, the names are also clear and not misleading!

ahmadsyuaib replied:
Thank you!

KennyLewi reviewed (commented):
LGTM! Except for some nits. I like how consistent you are with following the naming conventions specified in the course, this makes reading your code very easy 😄

ahmadsyuaib committed:
Correct codestyle

ahmadsyuaib committed:
Merge branch 'branch-Checkstyle'

* branch-Checkstyle:
  Correct codestyle

ahmadsyuaib committed:
Add GUI

ahmadsyuaib committed:
Clean code

ahmadsyuaib committed:
Make closing smoother

ahmadsyuaib committed:
Clean up GUI

ahmadsyuaib committed:
Merge branch 'branch-GUI'

* branch-GUI:
  Clean up GUI
  Make closing smoother
  Clean code
  Add GUI

ahmadsyuaib committed:
Fix array out of bounds error

ahmadsyuaib committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Fix array out of bounds error

ahmadsyuaib committed:
Update README file

ahmadsyuaib committed:
Clean code according to proper styling conventions

ahmadsyuaib committed:
Add assertion

Stobberi should only run if the variable isExit() method returns false. Otherwise, the programme should have terminated.

Let's add an assertion to check if the programme should be running at that point of time.

Adding an assertion will ensure that Stobberi only runs when the method returns false, preventing any potential problems in the future.

ahmadsyuaib committed:
Add assertion

Command should be a String for the programme to work. Or else, the methods that are going to be called will not work.

Let's add an assertion to check if variable "command" is a String.

Adding an assertion will ensure that the command variable has a String, preventing any potential problems in the future.

ahmadsyuaib committed:
Clean the code to remove if statement

TaskList has a function filterListByDate() which has an if statement that checks if the event is during the date. However, this is not necessary.

Let's remove this if statement.

By removing this if statement, we are able to shorten the length of our method and also avoid deep nesting.

ahmadsyuaib committed:
Merge pull request #1 from ahmadsyuaib/branch-A-Assertions

Branch a assertions

ahmadsyuaib committed:
Merge pull request #2 from ahmadsyuaib/branch-A-CodeQuality

Clean the code to remove if statement

ahmadsyuaib committed:
Add isSame method

ahmadsyuaib committed:
Add hasTask method

ahmadsyuaib committed:
Add exception for when task is repeated

ahmadsyuaib committed:
Create a new Command that displays Help message

ahmadsyuaib committed:
Add another case for help

ahmadsyuaib committed:
Edit greeting message

ahmadsyuaib committed:
Edit the error message to give it character

ahmadsyuaib committed:
Edit the error message to make it more obvious

ahmadsyuaib committed:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Edit the error message to make it more obvious
  Edit the error message to give it character

ahmadsyuaib committed:
Extract error thrown to make the code cleaner

ahmadsyuaib committed:
Throw an error

ahmadsyuaib committed:
Give personality to the chatbot

ahmadsyuaib committed:
Throw an error

ahmadsyuaib committed:
Give obvious personality for errors thrown

ahmadsyuaib committed:
Merge branch 'branch-A-Personality'

* branch-A-Personality:
  Give obvious personality for errors thrown
  Throw an error
  Give personality to the chatbot
  Throw an error
  Extract error thrown to make the code cleaner

ahmadsyuaib committed:
Add error handling for extra input in command

ahmadsyuaib committed:
Add new argument into ExitCommand instance

ahmadsyuaib committed:
Format the parser

ahmadsyuaib committed:
Add error handling to handle more than 1 input

ahmadsyuaib committed:
Add error handling for invalid numbers

ahmadsyuaib committed:
Create new class for invalid inputs

ahmadsyuaib committed:
Edit the javadoc

ahmadsyuaib committed:
Add error handling for irregular spacing

ahmadsyuaib committed:
Update the filer with new methods

ahmadsyuaib committed:
Update javadocs accordingly

ahmadsyuaib committed:
Update javadocs of isSame method

ahmadsyuaib committed:
Add error handling for multiple inputs

ahmadsyuaib committed:
Abstract taskList and restOfCommand to make cleaner code

ahmadsyuaib committed:
Add error handling for extra inputs

ahmadsyuaib committed:
Change greetings

ahmadsyuaib committed:
Clean code to make it neater

ahmadsyuaib committed:
Add error handling for cases where there is no tasks

ahmadsyuaib committed:
Merge branch 'branch-A-MoreErrorHandling'

* branch-A-MoreErrorHandling:
  Add error handling for cases where there is no tasks
  Clean code to make it neater
  Change greetings
  Add error handling for extra inputs
  Abstract taskList and restOfCommand to make cleaner code
  Add error handling for multiple inputs
  Update javadocs of isSame method
  Update javadocs accordingly
  Update the filer with new methods
  Add error handling for irregular spacing
  Edit the javadoc
  Create new class for invalid inputs
  Add error handling for invalid numbers
  Add error handling to handle more than 1 input
  Format the parser
  Add new argument into ExitCommand instance
  Add error handling for extra input in command

ahmadsyuaib committed:
Improve javadocs of all the methods

ahmadsyuaib committed:
Create new test file for Storage class

ahmadsyuaib committed:
Add comprehensive testing using ChatGPT

ahmadsyuaib committed:
Add comprehensive testing using ChatGPT

ahmadsyuaib committed:
Create Task as abstract class

ahmadsyuaib committed:
Merge branch 'branch-A-AiAssisted'

* branch-A-AiAssisted:
  Create Task as abstract class
  Add comprehensive testing using ChatGPT
  Add comprehensive testing using ChatGPT
  Create new test file for Storage class
  Improve javadocs of all the methods

ahmadsyuaib committed:
Add comprehensive testing for the different methods

ahmadsyuaib committed:
Merge branch 'branch-A-MoreTesting'

* branch-A-MoreTesting:
  Add comprehensive testing for the different methods

ahmadsyuaib committed:
Add icon and application name onto windows title bar

ahmadsyuaib committed:
Change the display time format to show minutes

ahmadsyuaib committed:
Change greeting

ahmadsyuaib committed:
Clean code according to course style guide

ahmadsyuaib committed:
Update main class set to correct launcher file

ahmadsyuaib committed:
Add representative screenshot of the product

ahmadsyuaib committed:
Add a user guide

ahmadsyuaib committed:
Format picture and link to jar file

ahmadsyuaib committed:
Update picture formatting
